File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/OrganizationSubmitEventDto.java
package de.helfenkannjeder.helfomat.api.organisation;

import de.helfenkannjeder.helfomat.api.organisation.event.OrganizationEventDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

import java.util.List;

/**
 * @author Valentin Zickner
 */
public class OrganizationSubmitEventDto {

    private OrganisationId organizationId;
    private String sources;
    private List<OrganizationEventDto> events;

    private OrganizationSubmitEventDto() {
    }

    public OrganizationSubmitEventDto(OrganisationId organizationId, String sources, List<OrganizationEventDto> events) {
        this.organizationId = organizationId;
        this.sources = sources;
        this.events = events;
    }

    public OrganisationId getOrganizationId() {
        return organizationId;
    }

    public String getSources() {
        return sources;
    }

    public List<OrganizationEventDto> getEvents() {
        return events;
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/OrganizationSubmitEventDto.kt
package de.helfenkannjeder.helfomat.api.organization

import de.helfenkannjeder.helfomat.api.organization.event.OrganizationEventDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationSubmitEventDto(
    val organizationId: OrganizationId,
    val sources: String,
    val events: List<OrganizationEventDto>
)